From a4c505582e706dcefafd7169e11c603642c85eab Mon Sep 17 00:00:00 2001
From: Dheeraj CVR <cvr.dheeraj@gmail.com>
Date: Sat, 19 Apr 2014 02:33:46 +0530
Subject: [PATCH 104/206] Camera2: Prevent autofocus when video snapshot is in
 progress

During video recording if screen is tapped repeatedly, autofocus is
started while the snapshot is taken. This would either result in an
out of focus snapshot which would also disturb the focus in the recorded
video or crash the sensors that can't handle autofocus during snapshot.

Change-Id: If2763094f5ab7536e4409469276de54580e803c8
---
 src/com/android/camera/VideoModule.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/com/android/camera/VideoModule.java b/src/com/android/camera/VideoModule.java
index e2bfffc4b..645eb4446 100755
--- a/src/com/android/camera/VideoModule.java
+++ b/src/com/android/camera/VideoModule.java
@@ -642,7 +642,7 @@ public class VideoModule implements CameraModule,
             }
 
             // Check if metering area or focus area is supported.
-            if (mFocusAreaSupported || mMeteringAreaSupported) {
+            if ((mFocusAreaSupported || mMeteringAreaSupported) && !mSnapshotInProgress) {
                 mFocusManager.onSingleTapUp(x, y);
             }
         }
-- 
2.17.1

